spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/config/EnvironmentMeterFilter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.metrics.config;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Function;
+
+import io.micrometer.core.instrument.config.PropertyMeterFilter;
+
+import org.springframework.core.convert.ConversionFailedException;
+import org.springframework.core.convert.support.DefaultConversionService;
+import org.springframework.core.env.Environment;
+
+
+/**
+ * Meter filter constructed from properties.
+ *
+ * @author Jon Schneider
+ */
+public class EnvironmentMeterFilter extends PropertyMeterFilter {
+	private final Environment environment;
+	private final DefaultConversionService conversionService = new DefaultConversionService();
```

Comment:
Is there a better way of accessing conversions in Boot 2?

Reply:
I think it's a bean and you can inject one. See [here](https://github.com/spring-cloud/spring-cloud-netflix/blob/c6fa1627280eec392ee6043d9935753b30e1a639/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/support/SpringMvcContract.java#L87) and [the auto-configuration](https://github.com/spring-cloud/spring-cloud-netflix/blob/c6fa1627280eec392ee6043d9935753b30e1a639/spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientsConfiguration.java#L82)

Reply:
@spencergibb In a vanilla Spring Boot 1.5.x app, it is not injectable. Must be configured somewhere.